cli/commands.go

```diff
@@ -76,3 +91,40 @@ var get = &cobra.Command{
 		return nil
 	},
 }
+
+var server = &cobra.Command{
+	Use:   "server",
+	Short: "starts server",
+	Long:  "give 2 parameters: port and filepath",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cReader := &constReader{}
+		port, err := cmd.Flags().GetString("port")
+		if err != nil {
+			return err
+		}
+		path, err := cmd.Flags().GetString("path")
+		if err != nil {
+			return err
+		}
+		secretService := service.New(storage.NewFsStorage(path),
+			ciphering.NewAESEncoder(ciphering.NewRandomNonceProducer(rand.Reader)),
+			ciphering.NewAESEncoder(ciphering.NewRandomNonceProducer(cReader)))
+
+		sign := make(chan os.Signal, 1)
+		signal.Notify(sign, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+
+		srv := rest.NewSecretRestAPI(secretService, port)
+		serverCtx, serverStopCtx := context.WithCancel(context.Background())
+		go func() {
+			<-sign
+			log.Println("got signal")
+
+			srv.Stop()
+			serverStopCtx()
+		}()
+		err = srv.Start(serverCtx)
+		<-serverCtx.Done()
+		log.Println("Done")
```

Comment:
```suggestion
		go func() {
			defer signal.Stop(sign)
                        defer serverCancel()
                        
                        select {
                        case <-sign:
                        case <-serverCtx.Done():
                        }
		}()
		err = srv.Start(serverCtx)
		log.Println("Done")
```

It will be more incapsulated and less error prone  if shutdown is written close to the place where it knows how to stop itself (mean in `src.Start` method).